tap_google_sheets/transform.py

```diff
@@ -145,37 +145,77 @@ def transform_sheet_boolean_data(value, unformatted_value, sheet_title, col_name
         return col_val
 
 # transform decimal values in the sheet
-def transform_sheet_decimal_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+def transform_sheet_decimal_data(value_type, formatted_value, unformatted_value, sheet_title, col_name, col_letter, row_num, col_type):
+    """
+        Transform number type data and return according to the datatype in the sheet
+
+        :param
+        value_type: the datatype of unformatted value ie. int or float
+        formatted_value - The displayed value of a cell in the sheet
+        unformatted_value - The formatted value of a cell in the sheet
+        sheet_title - The title of the sheet
+        col_name - Column name
+        col_letter - Column letter of the record ie. A, B, C, etc.
+        row_num - Row number of the record
+        col_type - Column type of the record (here: numberType)
+    """
+
+    # Removing comma to handle US number type format ie. 123,456.10 -> 123456.10
+    numeric_value = formatted_value.replace(",", "")
+    try:
+        # Verify we can convert formatted value to float for scientific formatted numbers
+        # For example:
+        #   formatted value: "1.23E+03"
+        #   unformatted value: 1234
+        # thus, we can convert "1.23E+03" to float but, for int casting we get error and wrong value will be returned
+        float(numeric_value)
+    except ValueError:
+        return str(formatted_value) # Return original value in case of ValueError
```

Comment:
@harshpatel4crest can you add a logger debug message here that "Received the cell-value '<formatted_value>' with unexpected data type for Column X of sheet Y. Ingesting this value with string format in the target"